### ISSUE TITLE
Discuss and possibly implement `--interactive` CLI arguments

### ISSUE THREAD

**Issue by andrewtavis** [scribe-org/Scribe-Data#158]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/scribe-org/Scribe-Data/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow Scribe-Data's [Code of Conduct](https://github.com/scribe-org/Scribe-Data/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

This issue is first a discussion and then potentially an implementation of an `--interactive` (`-i`) command for the Scribe-Data CLI. This might help users more easily explore and onboard to the CLI by providing options that they can choose from :)

It could work something like:

```
scribe-data query -i

Language options:
--------------------------------
| 1. English | 2. French | ... |
--------------------------------

Please enter the languages to query data for, their numbers or (a) for all languages: a

Word type options:
-----------------------------
| 1. nouns | 2. verbs | ... |
-----------------------------

Please enter the word types to query data for, their numbers or (a) for all word types: 1
File type to export (json, csv, tsv, sqlite) [json]: json
Export directory path [./scribe_data_json_export]:
Overwrite existing data without asking (y, n) [n]: y

Querying nouns for all languages that will be exported as JSONs to './scribe_data_json_export' ...
```

How the above would work for `list` is a question 🤔

### Contribution

Happy to discuss and eventually help implement this! Options would basically come from `list` commands and command arguments :)

**Comment by andrewtavis** [comment]
CC @wkyoshida and @mhmohona :) Let me know what you think of the above!

**Comment by andrewtavis** [comment]
Note from the meeting just now, we wouldn't need this to just be `scribe-data q -i`, but we could just do `scribe-data -i` :)

**Comment by andrewtavis** [comment]
CC @mhmohona for working on this, please write in here :)

**Comment by mhmohona** [comment]
✨

**Comment by mhmohona** [comment]
@andrewtavis, what do you think? 
![image](https://github.com/scribe-org/Scribe-Data/assets/14244685/f6b05c9b-6be8-42b0-92ed-b71dc117ab19)

**Comment by andrewtavis** [comment]
This is looking amazing, @mhmohona! I think that we will need to eventually shift over to have a grid of selections as when we have 20-30 languages a list won't be the best user experience, but then we can make a separate issue for this once this is merged 😊

**Comment by wkyoshida** [comment]
Hey! :wave: 
Not sure if what you had in mind @andrewtavis was something along the lines of a Terminal UI kinda deal with the `--interactive` option. Something like the image below:

![image](https://github.com/scribe-org/Scribe-Data/assets/15043193/b97454d3-ca1c-48be-a577-5aa385a5e60b)

where the CLI takes up the terminal with its interface, but could be an idea on how to implement this as well :thinking: Would be cool! But we can discuss how much of this we'd want to actually include within GSoC.

Anyways - looks like we would be able to do this if we learn some [curses](https://docs.python.org/3/library/curses.html) :hand_over_mouth: 
Looks like this is a library that would enable us to do this. Refs:
- [Instructions on how to curse](https://docs.python.org/3/howto/curses.html)
- [StackOverflow - console application gui for python](https://stackoverflow.com/questions/12664184/console-application-gui-for-python)

**Comment by andrewtavis** [comment]
This is really interesting, @wkyoshida! We could maybe look into this after GSoC, with what @mhmohona has for #167 being the majority of what we get done for the program?

**Comment by andrewtavis** [comment]
Closed via #167 😊 We can add to this as we go, and there seem to be some minor bugs that we can figure out before the v4.0 release, but great progress here!
